src/Protocols/Plugin.hs

```diff
@@ -2,10 +2,26 @@
 A GHC source plugin providing a DSL for writing Circuit components. Credits to
 @circuit-notation@ at <https://github.com/cchalmers/circuit-notation>.
 -}
-{-# LANGUAGE OverloadedStrings #-}
 {-# LANGUAGE CPP #-}
+{-# LANGUAGE OverloadedStrings #-}
+{-# LANGUAGE PatternSynonyms #-}
+
+module Protocols.Plugin
+  ( plugin
+  , circuit
+  , (-<)
+
+  -- XXX: Haddock will try to link references even for non-exported identifiers. We
```

Comment:
I usually prefer [this](https://stackoverflow.com/questions/16887023/is-it-possible-to-hide-specific-functions-from-appearing-in-the-documentation-us) solution for this case, which is clean and transparent.

Reply:
Done!